### PR TITLE
Handle missing precinct data for older Dominion data sets #533

### LINKED ISSUE #533: Support older Dominion data sets
Older (pre 2020) Dominion data sets might not have precinct data.  For example, SF 2019 consolidated: https://sfelections.sfgov.org/november-5-2019-election-results-detailed-reports

In this case we need to handle missing precinct data gracefully.

### PR DESCRIPTION
**Pull request by moldover**
Fixes #533

### CODE REVIEW

File: src/main/java/network/brightspots/rcv/DominionCvrReader.java [hunk 1]
```diff
@@ -208,13 +215,17 @@ private void parseCvrFile(
...
+        if (precinctId != null && (this.precincts == null || !this.precincts
+            .containsKey(precinctId))) {
           Logger.severe("Precinct ID \"%d\" from CVR not found in manifest data!", precinctId);
           throw new CvrParseException();
         }
-        String precinct = this.precincts.get(precinctId);
+        String precinct = null;
+        if (this.precincts != null) {
+          precinct = this.precincts.get(precinctId);
+        }
```
Comment (HEdingfield) [inline src/main/java/network/brightspots/rcv/DominionCvrReader.java:228]: Nit: you could simplify this a bit:
`String precinct = this.precincts != null ? this.precincts.get(precinctId) : null;`

**Review by HEdingfield — COMMENTED** [review]
Would you or the person who provided you with the data be able to cut 99% of the data out of the data set so it's closer to 5 MB? As it stands, it's not an official regression test because there's no inclusion in `TabulatorTests.java`, and it seems like it should be.

File: src/test/resources/network/brightspots/rcv/test_data/dominion_no_precinct_data/dominion_no_precinct_data_summary.json [hunk 2]
```diff
@@ -0,0 +1,163 @@
+{
```
Comment (HEdingfield) [inline src/test/resources/network/brightspots/rcv/test_data/dominion_no_precinct_data/dominion_no_precinct_data_summary.json:1]: I don't think this file is necessary, right?

File: src/test/resources/network/brightspots/rcv/test_data/dominion_no_precinct_data/dominion_no_precinct_data/BallotTypeContestManifest.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (HEdingfield) [inline src/test/resources/network/brightspots/rcv/test_data/dominion_no_precinct_data/dominion_no_precinct_data/BallotTypeContestManifest.json:1]: Nit: going off the directory structure pattern for the other dominion tests, it'd be more consistent to name the folders:
`/dominion_no_precinct/dominion_no_precinct_input_data`

**Review by HEdingfield — APPROVED** [review]
Great, glad that worked out!

File: src/test/resources/network/brightspots/rcv/test_data/dominion_no_precinct_data/dominion_no_precinct_data_summary.json (hunk 2 above)
Comment (moldover) [reply src/test/resources/network/brightspots/rcv/test_data/dominion_no_precinct_data/dominion_no_precinct_data_summary.json:1]: ack!

**Pull request merged**